python/ray/serve/http_proxy.py

```diff
@@ -26,11 +26,21 @@ class HTTPProxy:
     # blocks forever
     """
 
-    def __init__(self):
+    async def fetch_config_from_master(self):
         assert ray.is_initialized()
-        master = ray.util.get_actor(SERVE_MASTER_NAME)
-        self.route_table, [self.router_handle] = ray.get(
-            master.get_http_proxy_config.remote())
+
+        from ray.serve.api import init, _get_master_actor
+        init()
+        master = _get_master_actor()
+        self.route_table, [self.router_handle
+                           ] = await master.get_http_proxy_config.remote()
+        [self.metric_sink] = await master.get_metric_sink.remote()
+
+        self.metric_collector = PushCollector.connect_from_serve()
+        self.request_counter = self.metric_collector.new_counter(
+            "num_http_requests",
+            description="The number of requests processed",
+        )
```

Comment:
These should probably go in `__init__` (unless this needs to be in an async context, in which case we should rename this method)